src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -541,6 +549,9 @@ else if (isBigDecimal())
 
     public ByteSource asComparableBytes(ByteBuffer value, ByteComparable.Version version)
     {
+        if (value.remaining() == 0)
+            return ByteSource.EMPTY;
```

Comment:
boolean and a couple other types will return `null` from their implementations of `asComparableBytes()` if we don't do this

Reply:
is this safe though in general?  shouldn't we check for the expected types?

Reply:
Possibly...I'll think this one over a bit. Keep in mind that these values only have to be consistent in index space. i.e. As long as this is the value queries would use for empty, it should be fine.